### PR TITLE
feat: Add zero address `receiver` checks to `_mint` and `_burn` (SC-5371)

### PR DESCRIPTION
**Pull request by edag94**
require(receiver_ != address(0), "RDT:M:ZERO_RECEIVER")

### CODE REVIEW

File: contracts/RevenueDistributionToken.sol [hunk 1]
```diff
@@ -162,8 +163,9 @@ contract RevenueDistributionToken is IRevenueDistributionToken, ERC20 {
     }
 
     function _burn(uint256 shares_, uint256 assets_, address receiver_, address owner_, address caller_) internal {
-        require(shares_ != 0, "RDT:B:ZERO_SHARES");
-        require(assets_ != 0, "RDT:B:ZERO_ASSETS");
+        require(receiver_ != address(0), "RDT:M:ZERO_RECEIVER");
```
Comment (vbidin) [inline contracts/RevenueDistributionToken.sol:166]: Would it make sense to also check if the `owner_` is set to `address(0)` here?

**Review by vbidin — DISMISSED** [review]

File: contracts/test/accounts/Staker.sol [hunk 2]
```diff
@@ -13,24 +13,32 @@ contract Staker is ERC20User {
...
+        assets_ = IRDT(token_).mint(shares_, receiver_);
+    }
+
     function rdToken_redeem(address token_, uint256 shares_) external returns (uint256 assets_) {
         assets_ = IRDT(token_).redeem(shares_, address(this), address(this));
     }
 
     function rdToken_redeem(address token_, uint256 shares_, address recipient_, address owner_) external returns (uint256 assets_) {
-        return IRDT(token_).redeem(shares_, recipient_, owner_);
+        assets_ = IRDT(token_).redeem(shares_, recipient_, owner_);
```
Comment (lucas-manuel) [inline contracts/test/accounts/Staker.sol:33]: Nice

File: contracts/RevenueDistributionToken.sol (hunk 1 above)
Comment (lucas-manuel) [reply contracts/RevenueDistributionToken.sol:166]: `RDT:B:ZERO_RECEIVER`

File: contracts/test/RevenueDistributionToken.t.sol [hunk 3]
```diff
@@ -559,6 +559,17 @@ contract DepositFailureTests is RDTTestBase {
         staker = new Staker();
     }
 
+    function test_deposit_zeroReceiver() public {
+
```
Comment (lucas-manuel) [inline contracts/test/RevenueDistributionToken.t.sol:563]: Rm extra line

File: contracts/test/RevenueDistributionToken.t.sol [hunk 4]
```diff
@@ -1468,6 +1479,17 @@ contract MintFailureTests is RDTTestBase {
         staker = new Staker();
     }
 
+        function test_mint_zeroReceiver() public {
```
Comment (lucas-manuel) [inline contracts/test/RevenueDistributionToken.t.sol:1482]: Fix indentation

File: contracts/test/RevenueDistributionToken.t.sol [hunk 5]
```diff
@@ -1468,6 +1479,17 @@ contract MintFailureTests is RDTTestBase {
...
+
```
Comment (lucas-manuel) [inline contracts/test/RevenueDistributionToken.t.sol:1483]: Rm extra line

File: contracts/test/mocks/MockRevertingERC20.sol [hunk 6]
```diff
@@ -58,6 +60,7 @@ contract MockRevertingERC20 {
     }
 
     function _transfer(address owner_, address recipient_, uint256 amount_) internal {
+        require(recipient_ != revertingDestination, "INVALID");
```
Comment (lucas-manuel) [inline contracts/test/mocks/MockRevertingERC20.sol:63]: I think we should just replace this function with `require(false, "MOCK_TRANSFER_REVERT");`

Comment (lucas-manuel) [reply contracts/test/mocks/MockRevertingERC20.sol:63]: And then we can get rid of the `revertingDestination`

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

**Review by lucas-manuel — APPROVED** [review]

**Pull request merged**